runners/spark/src/main/java/org/apache/beam/runners/spark/aggregators/NamedAggregators.java

```diff
@@ -39,170 +41,185 @@
  * is launched. We can then add aggregators on the fly in Spark.
```

Comment:
I find this comment confusing. It starts by saying that Spark expects the accumulators be declared before a job is launched, and ends by saying that aggregators can be added on the fly. Further, I believe that Beam currently requires aggregators to be pre-defined during pipeline construction.

Is there somewhere I can get an overview of accumulators in Spark and how they work in Beam? As Dan mentioned, I'm currently looking at whether we should move the existing Aggregator API towards a Metrics API that is more similar to something like dropwizard. It would be useful to understand how things work in Spark today.